BuildScripts/mvBESFunctions.sql

```diff
@@ -235,6 +236,93 @@ begin
 
+END;
+$BODY$
+LANGUAGE    plpgsql
+SECURITY    DEFINER;
+------------------------------------------------------------------------------------------------------------------------------------
+CREATE OR REPLACE
+FUNCTION    mv$renameMaterializedViewLog
+            (
+                pOldTableName           IN      TEXT,
+                pNewTableName           IN      TEXT,
+                pOwner                  IN      TEXT        DEFAULT USER
+            )
+    RETURNS VOID
+AS
+$BODY$
+/* ---------------------------------------------------------------------------------------------------------------------------------
+Routine Name: mv$renameMaterializedViewLog
+Author:       Ethan McQuarrie
+Date:         06/12/2021
+------------------------------------------------------------------------------------------------------------------------------------
+Revision History    Push Down List
+------------------------------------------------------------------------------------------------------------------------------------
+Date        | Name          | Description
+------------+---------------+-------------------------------------------------------------------------------------------------------
+            |               |
+------------+---------------+-------------------------------------------------------------------------------------------------------
+Description:    Renames a materialized view log table
+
+Arguments:      IN      pOldTableName       The name of the base table the materialized view logs are against
+                IN      pNewTableName       The name this table is being updated to
+                IN      pOwner              Optional, the owner of the materialized view, defaults to user
+Returns:                VOID
+
+************************************************************************************************************************************
+Copyright 2021 Beyond Essential Systems Pty Ltd
+***********************************************************************************************************************************/
+DECLARE
+
+    rConst          mv$allConstants;
+    aViewLog        pg$mview_logs;
+
+    tRenameLogTableSqlStatement             TEXT;
+    tUpdateMViewLogsTableSqlStatement       TEXT;
+    tUpdateMViewTableColumns                TEXT;
+BEGIN
+
+    rConst   := mv$buildAllConstants();
+    tRenameLogTableSqlStatement         := rConst.ALTER_TABLE || 'log$_' || pOldTableName || rConst.RENAME_TO_COMMAND || 'log$_' || pNewTableName;
+    tUpdateMViewLogsTableSqlStatement   := rConst.UPDATE_COMMAND || 'pg$mview_logs' || rConst.SET_COMMAND
+                                            || 'table_name = ' || rConst.SINGLE_QUOTE_CHARACTER || pNewTableName || rConst.SINGLE_QUOTE_CHARACTER || rConst.COMMA_CHARACTER
+                                            || 'pglog$_name = ' || rConst.SINGLE_QUOTE_CHARACTER || 'log$_' || pNewTableName || rConst.SINGLE_QUOTE_CHARACTER
```

Comment:
You're leaving the trigger with an inconsistent name by the looks of it? Would be nice to have that all match up, sure to confuse someone down the line